### DISCUSSION TITLE
Retrieve field data on vector search (C# SDK)

### DISCUSSION THREAD

**Discussion by jnprovence1** [milvus-io/milvus#29067] [Q&A and General discussion]
In the photo below, you will notice that I have a variable that I have pulled from the SearchResults called id_field.  It is of type FieldData.  In the window at the bottom, you can see there is a Data property that holds the integers I need (these are not the Milvus id's, they are id's for another system)

How in the world am I supposed to access these?  FieldData has no property called Data in the definition.  I feel like I must be overthinking this.

Thank you for any help!

![2023-12-07 15_19_58-Window](https://github.com/milvus-io/milvus/assets/8193574/d521465a-2f05-44f6-820b-89df2b37f123)

**Comment by yhmo** [comment]
`FieldData` is an abstract class: 
https://github.com/milvus-io/milvus-sdk-csharp/blob/c1d268949bb115715f1052219d88c2240b428fd4/Milvus.Client/FieldData.cs#L9


`FieldData<TData>` is a template class derived from `FieldData`, and this template class has the member "Data"
https://github.com/milvus-io/milvus-sdk-csharp/blob/c1d268949bb115715f1052219d88c2240b428fd4/Milvus.Client/FieldData.cs#L322

You can cast the `FieldData` to `FieldData<TData>` according to its type, then you can access the "Data" member. Just like the test code shows:
https://github.com/milvus-io/milvus-sdk-csharp/blob/c1d268949bb115715f1052219d88c2240b428fd4/Milvus.Client.Tests/SearchQueryTests.cs#L21
